### PR TITLE
test: create general purpose vscode operator function & finish select azure tenant test case

### PR DESCRIPTION
**Pull request by ayachensiyuan**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -425,6 +425,7 @@
     "precommit": "lint-staged"
   },
   "devDependencies": {
+    "@aws-sdk/client-s3": "^3.622.0",
```
Comment (wenytang-ms) [inline package.json:428]: what's this?

File: src/test/e2e/tests/registerApiViaCICD.test.ts [hunk 2]
```diff
@@ -1,37 +1,38 @@
...
 import { expect, test } from '../baseTest';
+import { APICenter, Timeout, VSCode } from '../utils/constants';
+import VscodeOperator from '../utils/vscodeOperator';
 
 test('trigger generateAPIviaCICD with Azure DevOps', async ({ workbox }) => {
+    await workbox.waitForTimeout(Timeout.PREPARE_TEST);
     // wait API Center extension installed on VS Code.
-    await workbox.getByRole("tab", { name: "API Center" }).isVisible();
-    await workbox.getByRole("tab", { name: "API Center" }).locator('a').click();
+    expect(await VscodeOperator.isSideTabItemExist(workbox, VSCode.TAB_API_CENTER)).toBeTruthy();
```
Comment (wenytang-ms) [inline src/test/e2e/tests/registerApiViaCICD.test.ts:11]: it's better to replace the VSCode.TAB_API_CENTER to specific string, we need to assert the workbox with the string, not config string assert to itselft.

File: src/test/e2e/tests/registerApiViaCICD.test.ts [hunk 3]
```diff
@@ -1,37 +1,38 @@
...
+    await VscodeOperator.activeSideTab(workbox, VSCode.TAB_API_CENTER);
     // trigger command palette.
-    await workbox.keyboard.press('Control+Shift+KeyP');
-    await workbox.getByRole("combobox", { name: "INPUT" }).fill('>Azure API Center: Register API');
-    await workbox.getByRole('listbox').first().press('Enter');
+    await VscodeOperator.execCommandInCommandPalette(workbox, APICenter.REGISTER_API);
     // select the first option.
-    await workbox.getByRole('option', { name: "CI/CD" }).locator('a').click();
+    await VscodeOperator.selectOptionByName(workbox, APICenter.CI_CD);
```
Comment (wenytang-ms) [inline src/test/e2e/tests/registerApiViaCICD.test.ts:16]: same here, all the target string need to consider as a direct string not from config.

**Review by wenytang-ms — APPROVED** [review]

File: src/test/e2e/tests/selectAzureTenant.test.ts [hunk 4]
```diff
@@ -1,10 +1,15 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
-import { test } from '../baseTest';
+
+import { expect, test } from '../baseTest';
+import { APICenter, TestENV, Timeout, VSCode } from '../utils/constants';
+import VscodeOperator from '../utils/vscodeOperator';
+
 test('select Tenant', async ({ workbox }) => {
```
Comment (formulahendry) [inline src/test/e2e/tests/selectAzureTenant.test.ts:8]: So this test case is not 'select Tenant' now? 
It is showing subs when there is only 1 tenant

Comment (ayachensiyuan) [reply src/test/e2e/tests/selectAzureTenant.test.ts:8]: Indeed it is. The step of selecting tenant is put in prepare step, and this case becomes to verify whether subs meets expectations. Can I change title to "showing subs"？

Comment (ayachensiyuan) [reply src/test/e2e/tests/selectAzureTenant.test.ts]: > So this test case is not 'select Tenant' now? It is showing subs when there is only 1 tenant

change case name to "validate subscription"

**Review by formulahendry — APPROVED** [review]

**Pull request merged**